src/caf/toolkit/config_base.py

```diff
@@ -220,6 +220,39 @@ def write_example(
 
 
 # # # FUNCTIONS # # #
+def _is_collection(object: any):
+    """
+    Check if an object is any type of non-dict collection.
+
+    Currently only checks for list, tuple or set,
+    """
+    if isinstance(object, list):
+        return True
+    elif isinstance(object, tuple):
+        return True
+    elif isinstance(object, set):
+        return True
+    elif isinstance(object, dict):
+        return True
+    return False
+
+
+def _remove_none_collection(data: Union[list, set, tuple]):
+    """Remove items recursively from collections which are None"""
+    # empty list, type set later
+    filtered = []
+    for item in data:
+        if item is None:
+            continue
+        elif isinstance(item, dict):
+            item = _remove_none_dict(item)
```

Comment:
Don't need the first `elif` if you're using `continue`

```suggestion
        if item is None:
            continue
           
        if isinstance(item, dict):
            item = _remove_none_dict(item)
```